src/test/java/com/gildedrose/GildedRoseTest.java

```diff
@@ -2,16 +2,160 @@
 
 import static org.junit.Assert.*;
 
+import org.junit.Before;
 import org.junit.Test;
 
 public class GildedRoseTest {
 
+    private static final String ITEM_DEFAULT_NAME = "foo";
+    private static final int ITEM_DEFAULT_SELLIN = 15;
+    private static final int ITEM_DEFAULT_QUALITY = 15;
+    private static final int ITEM_OVERDUE_SELLIN = 0;
+    private static final int ITEM_LONG_OVERDUE_SELLIN = -50;
+    private static final int ITEM_QUALITY_MAX = 50;
+    private static final String ITEM_NAME_BACKSTAGE_PASS = "Backstage passes to a TAFKAL80ETC concert";
+    private static final String ITEM_NAME_AGED_BRIE = "Aged Brie";
+    private static final String ITEM_NAME_SULFURAS = "Sulfuras, Hand of Ragnaros";
+    private static final int ITEM_QUALITY_MIN = 0;
+    private Item item;
+    private GildedRose app;
+
+    @Before
+    public void setUp() {
+        item = new Item(ITEM_DEFAULT_NAME, ITEM_DEFAULT_SELLIN, ITEM_DEFAULT_QUALITY);
+        final Item[] items = new Item[] {item};
+        app = new GildedRose(items);
+    }
+
+    @Test
+    public void doesNotAlterName() {
+        app.updateQuality();
+        assertEquals(ITEM_DEFAULT_NAME, app.items[0].name);
+    }
+
+    @Test
+    public void doesDecreaseSellIn() {
+        app.updateQuality();
+        assertEquals(ITEM_DEFAULT_SELLIN - 1, app.items[0].sellIn);
+    }
+
+    @Test
+    public void doesDecreaseQuality() {
+        app.updateQuality();
+        assertEquals(ITEM_DEFAULT_QUALITY - 1, app.items[0].quality);
+    }
+
+    @Test
+    public void withPassedSellIn_doesDecreaseQuality() {
+        item.sellIn = ITEM_OVERDUE_SELLIN;
+        app.updateQuality();
+        assertEquals(ITEM_DEFAULT_QUALITY - 2, app.items[0].quality);
```

Comment:
asserting on `ITEM_DEFAULT_QUALITY` here, although it is not visible in the test body that this has been set => consequence of using a setup method in combination with constants being used.